akka-docs/rst/java/code/docs/http/javadsl/server/directives/CustomHttpMethodExamplesTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2015-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package docs.http.javadsl.server.directives;
+
+import akka.NotUsed;
+import akka.actor.ActorSystem;
+import akka.event.LoggingAdapter;
+import akka.event.NoLogging$;
+import akka.http.javadsl.ConnectHttp;
+import akka.http.javadsl.Http;
+import akka.http.javadsl.ServerBinding;
+import akka.http.javadsl.model.*;
+import akka.http.javadsl.server.Route;
+import akka.http.javadsl.settings.ParserSettings;
+import akka.http.javadsl.settings.ServerSettings;
+import akka.http.javadsl.testkit.JUnitRouteTest;
+import akka.http.scaladsl.model.RequestEntityAcceptance;
+import akka.stream.Materializer;
+import akka.stream.javadsl.Flow;
+import org.junit.Test;
+
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutionException;
+
+import static akka.http.javadsl.model.HttpProtocols.HTTP_1_0;
+
+public class CustomHttpMethodExamplesTest extends JUnitRouteTest {
+
+  @Test
+  public void testComposition() throws InterruptedException, ExecutionException {
+    ActorSystem  system = system();
+    Materializer materializer = materializer();
+    LoggingAdapter loggingAdapter = NoLogging$.MODULE$;
+
+    int    port = 9090;
+    String host = "127.0.0.1";
+
+    //#customHttpMethod
+    HttpMethod BOLT =
+      HttpMethods.custom("BOLT", false, true, RequestEntityAcceptance.Expected$.MODULE$);
```

Comment:
So in HttpMethod we need:

```
    /**
     * Returns the entity acceptance level for this method.
     */
    public abstract akka.http.scaladsl.model.RequestEntityAcceptance requestEntityAcceptance();
    
    /**
     * Java API: Returns the entity acceptance level for this method.
     */
    public abstract akka.http.javadsl.model.RequestEntityAcceptance getRequestEntityAcceptance();
```

Since we can't break compat on that method (return type becoming more general is not compat).

For the:

```
public static HttpMethod custom(String value, boolean safe, boolean idempotent, akka.http.scaladsl.model.RequestEntityAcceptance requestEntityAcceptance) {
```

We can't change the signature, it would break anyone who was calling this method.
But since this is in `javadsl` `HttpMethods` we can get away with:

```scala
public static HttpMethod createCustom(String value, boolean safe, boolean idempotent, akka.http.javadsl.model.RequestEntityAcceptance requestEntityAcceptance) {
```

and `@Deprecate` the previous one.

Inside that method we can cast to `(akka.http.scaladsl.model.RequestEntityAcceptance)`, since we know the only implementations exist in scaladsl (all javadsl RequestEntityAcceptance ARE scaladsl instances).